scripts/format-staged.ps1

```
@@ -0,0 +1,22 @@
+<#
+Format only staged Rust files and re-add them to the index.
+Falls back to cargo fmt if rustfmt not present.
+#>
+$staged = git diff --name-only --cached --diff-filter=ACM
+$rs = $staged | Where-Object { $_ -match '\.rs$' }
+if (-not $rs) { Write-Host "No staged Rust files to format."; exit 0 }
+
+$fmtFound = (Get-Command rustfmt -ErrorAction SilentlyContinue) -ne $null
```

PSScriptAnalyzer (1.23.0)

[warning] $null should be on the left side of equality comparisons.

Suggested fix: Use $null on the left hand side for safe comparison with $null.

(PSPossibleIncorrectComparisonWithNull)